spark/v3.3/spark/src/main/scala/org/apache/spark/sql/execution/datasources/SparkExpressionConverter.scala

```diff
@@ -36,15 +35,30 @@ object SparkExpressionConverter {
     SparkFilters.convert(DataSourceStrategy.translateFilter(sparkExpression, supportNestedPredicatePushdown = true).get)
   }
 
-  @throws[AnalysisException]
-  def collectResolvedSparkExpression(session: SparkSession, tableName: String, where: String): Expression = {
+  def collectDeterministicSparkExpression(session: SparkSession,
```

Comment:
I am not sure I understand the purpose of this method and why we can't modify the existing method.

Reply:
`collectResolvedSparkExpressionOption` is used to distinguish unresolvedSparkExpression and  a derterministic expression (true or false).

after we got None from `collectResolvedSparkExpressionOption`, we apply `collectDeterministicSparkExpression` to the where statement to decide its value should be true or false. 

yes. it would be nice if we could combine these two functions together. 
but I don't have an idea now .